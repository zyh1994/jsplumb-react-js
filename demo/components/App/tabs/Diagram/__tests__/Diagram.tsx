import {shallow} from 'enzyme';
import React from 'react';
import Diagram from '../Diagram';

test('Home content', () => {

  const defaultFunc = () => { return; };

  const component = shallow(
    <Diagram
      connections={[]}
      diagramId='DAG'
      maxScale={2}
      minScale={0.25}
      nodes={{}}
      onAddConnection={defaultFunc}
      onRemoveConnection={defaultFunc}
      onRemoveNode={defaultFunc}
      onSelectionUpdate={defaultFunc}
      onSetDiagramScale={defaultFunc}
      onSetDiagramXYOffset={defaultFunc}
      onUpdateNode={defaultFunc}
      scale={1}
      selections={[]}
      xOffset={0.0}
      yOffset={0.0}
    />
  );
  expect(component).toMatchSnapshot();
});
