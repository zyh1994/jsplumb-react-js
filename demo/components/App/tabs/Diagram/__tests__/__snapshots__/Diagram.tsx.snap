// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home content 1`] = `
<div
  className="jspr-canvas"
>
  <AutoSizer
    disableHeight={false}
    disableWidth={false}
    onResize={[Function]}
    style={Object {}}
  >
    <Component />
  </AutoSizer>
  <Graph
    bridge={[Function]}
    className=""
    connections={Array []}
    height={500}
    id="DAG"
    maxScale={2}
    minScale={0.25}
    onAddConnection={[Function]}
    onPanEnd={[Function]}
    onRemoveConnection={[Function]}
    onSelect={[Function]}
    onZoom={[Function]}
    passOnProps={false}
    renderOnChange={false}
    scale={1}
    scaleFactor={1.224744871391589}
    settings={Object {}}
    style={Object {}}
    styleName=""
    width={500}
    xOffset={0}
    yOffset={0}
  />
</div>
`;
